File: docs/examples/REST/c/demo_info.c
#include "api_gateway_client.h"

#include <stdio.h>
#include <stdlib.h>

/* Returns 0 on success, 1 if the request failed (so callers can propagate
 * a non-zero process exit status, mirroring the Python example).
 */
static int print_endpoint(const ApiGatewayClient *client, const char *path) {
    char *body = api_gateway_get(client, path);
    printf("\n%s\n", path);
    if (body == NULL) {
        fprintf(stderr, "request failed: %s\n", path);
        return 1;
    }
    printf("%s\n", body);
    free(body);
    return 0;
}

int main(void) {
    const char *key = getenv("EDUMATCHER_API_KEY");
    if (key == NULL) {
        key = "key-trader-demo";
    }
    ApiGatewayClient client = api_gateway_client("127.0.0.1", 8080, key);

    int failed = 0;
    failed |= print_endpoint(&client, "/api/v1/status");
    failed |= print_endpoint(&client, "/api/v1/symbols");
    failed |= print_endpoint(&client, "/api/v1/session");
    return failed ? 1 : 0;
}

File: docs/examples/REST/python/demo_info.py
"""Print simple API Gateway status information to the terminal."""

from __future__ import annotations

import json
import os

from api_gateway_client import ApiGatewayClient


def main() -> None:
    base_url = os.environ.get("EDUMATCHER_API_URL", "http://127.0.0.1:8080")
    api_key = os.environ.get("EDUMATCHER_API_KEY", "key-trader-demo")
    client = ApiGatewayClient(base_url, api_key)
    for path in ("/api/v1/status", "/api/v1/symbols", "/api/v1/session"):
        print(f"\n{path}")
        print(json.dumps(client.get_json(path), indent=2, sort_keys=True))


if __name__ == "__main__":
    main()